src/main/java/spoon/reflect/declaration/CtTypeMember.java

```diff
@@ -30,6 +30,11 @@
 	 * @return declaring class
 	 */
 	@DerivedProperty
-	CtType<?> getDeclaringType();
+	<T> CtType<T> getDeclaringType();
 
+	/**
+	 * Returns top level declaring type of this. If this is already a TopLevelType, then it returns this
```

Comment:
Returns the top level type declaring this type if an inner type. If this is already a top-level type, then returns itself.